components/brave_shields/browser/ad_block_component_service_manager.cc

```diff
@@ -251,6 +262,37 @@ void AdBlockComponentServiceManager::EnableFilterList(const std::string& uuid,
   UpdateFilterListPrefs(uuid, enabled);
 }
 
+void AdBlockComponentServiceManager::CheckAdBlockComponentsUpdate() {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+
+  auto runner = base::SequencedTaskRunner::GetCurrentDefault();
+
+  runner->PostDelayedTask(FROM_HERE, base::BindOnce([]() {
+                            BraveOnDemandUpdater::GetInstance()->OnDemandUpdate(
+                                kAdBlockFilterListCatalogComponentId);
+                          }),
+                          base::Seconds(base::RandInt(0, 10)));
+
+  runner->PostDelayedTask(FROM_HERE, base::BindOnce([]() {
```

Comment:
Could you please optimize identical blocks here?
There is 3 about the same `runner->PostDelayedTask` in the method. I hope we could merge it in the one.